src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1304,6 +1311,97 @@ protected override void HandlePauseRequestAsync(IRequestResponder<PauseArguments
             m_program.CauseBreak();
             responder.SetResponse(new PauseResponse());
         }
+        
+        protected override void HandleGotoRequestAsync(IRequestResponder<GotoArguments> responder)
+        {
+            var response = new GotoResponse();
+            if (!m_isStopped)
+            {
+                responder.SetResponse(response);
+                return;
+            }
+
+            var builder = new ErrorBuilder(() => AD7Resources.Error_UnableToSetNextStatement);
+            try
+            {
+                if (m_gotoCodeContexts.TryGetValue(responder.Arguments.TargetId, out IDebugCodeContext2 gotoTarget))
+                {
+                    IDebugThread2 thread = null;
+                    lock (m_threads)
+                    {
+                        if (!m_threads.TryGetValue(responder.Arguments.ThreadId, out thread))
+                            throw new AD7Exception("Unknown thread id: " + responder.Arguments.ThreadId.ToString(CultureInfo.InvariantCulture));
+                    }
+                    BeforeContinue();
+                    builder.CheckHR(thread.SetNextStatement(null, gotoTarget));
+                }
```

Comment:
>} [](start = 16, length = 1)

I believe we also need to send a Stopped event with reason 'Goto'.

Reply:
True.
> The debug adapter first sends the response and then a ‘stopped’ event with reason ‘goto’.

Reply:
> But what if something goes wrong, do you still send an event?
I don't think you should need to.